DIPs/DIP1xxx-SSS.md

```diff
@@ -0,0 +1,184 @@
+the compiler try to move such a struct.
+
+The documentation for `opPostMove` MUST also emphasize that while manipulating the memory at the `opPostMove` source location is allowed,
+the memory will be effectively freed with no destruction immediately after the function's return. Implementors SHOULD be
+encouraged to define the argument to `opPostMove` to be `const ref` to gain some compiler protection against accidental manipulation.
+This does not harm the implementor's access to the data, as she already has a copy at the destination location.
+
+### Code emitted by the compiler on move
+
+When moving a struct, the compiler should call `__move_post_blt` on the struct, giving it both new and old instances.
+
+### `opPostMove` Decoration Considerations
+
+Ideally, `opPostMove` should be `@nogc`, `nothrow` and either `@safe` or `@trusted`. If that doesn't happen, trying to compile
+code that moves a struct from context that is `nothrow`, `@nogc` or `@safe` would result in a compilation error.
+
+We could force these by decorating `__move_post_blt` itself as `nothrow @nogc @safe`, thus not allowing `opPostMove` to be defined
+any other way (we cannot otherwise force `opPostMove` to be defined any specific way, as it is being defined by the user). I chose
+not to do so because the user might opt not to use, e.g., `@safe` anywhere in her program. It would, therefor, not make sense
+to force her to use it in `opPostMove`. Due to attribute inference on template functions, if all member `opPostMove`s are, e.g., `@nogc`,
+D will automatically define `__move_post_blt` as `@nogc`.
+
+We do force `nothrow` on `opPostMove`, because throwing would mean a change in the program flow from a place that does not seem to
+run code, which might prove too confusing.
+
+## Performance Considerations
+
+Structs that do not define `opPostMove`, and that none of their members define `opPostMove`, will have their `__move_post_blt`
+implementation be just a function calling a bunch of empty functions recursively. Hopefully, the compiler will be able to inline
+this series of calls into oblivion, meaning the run time cost of this feature for structs that do not use it will be zero.
+
+If the compiler implementers fear that inlining will not nullify these calls where not applicable, they MAY manually eliminate
+no-op sub-trees. This may be done as simply as adding, at the beginning of `__move_post_blt`:
+
+```D
+static if( !hasElaborateMove!S )
+    return;
+```
+
+Such an addition incures some compile-time cost in the case that an inner member does have `opPostMove` defined, as it performs
+multiple scans of the subtrees during the recursive descent. It does, however, guarantee zero run time cost.
+
+Structs that do define `opPostMove` manage their own costs.
+
+## Effect on Phobos
+
+For the most part, no effect should happen on Phobos. Even if `opPostMove` is defines for a struct, the compiler's handling should
+make sure Phobos is not affected.
```

Comment:
"Even if `opPostMove` is defines for a struct, the compiler's handling should make sure Phobos is not affected."

This needs some clarity. What does "handling" mean? What is the compiler supposed to do here? Alternatively, just leave it out. You outline specific changes in Phobos below. In that case, replace "The exceptions are" with something like, "The required changes are limited to"